app/controllers/dradis/plugins/csv/upload_controller.rb

```diff
@@ -2,24 +2,44 @@ module Dradis::Plugins::CSV
   class UploadController < ::AuthenticatedController
     include ProjectScoped
 
-    before_action :load_attachment, only: [:new]
+    before_action :load_attachment, only: [:new, :create]
 
     def new
       @default_columns = ['Unique Identifier', 'Column Header From File', 'Type', 'Field in Dradis']
 
       @headers = ::CSV.open(@attachment.fullpath, &:readline)
+
+      @last_job = Log.new.uid
     end
 
     def create
-      redirect_to main_app.project_upload_manager_path(current_project)
+      job_logger.write 'Enqueueing job to start in the background.'
+
+      MappingImportJob.perform_later(
+        file: @attachment.fullpath.to_s,
+        identifier_col_index: mappings_params[:identifier],
```

Comment:
Too wordy for a param. What about `:identifier`, `:id_index`, `:id_col`? `id` is already used as the parameter for the content service so I think we can use that.